x/mongo/driver/operation.go

```diff
@@ -382,6 +389,19 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 		conn = nil
 	}
 
+	wm := memoryPool.Get().(*[]byte)
```

Comment:
Does using a `*[]byte` actually change the memory use behavior? It seems like using a `[]byte` should behave exactly the same with the current code in `Execute`.

Reply:
They do behave exactly the same. The purpose of using a pointer is to satisfy the linter and follow the idiom of using a pool.

Reply:
Ah, I see the `staticcheck` linter complains about putting non-pointer types into a `sync.Pool`. I think I understand the intent here now.

Reply:
Optional: Consider leaving a comment about why we use a `*[]byte` here instead of just `[]byte`. It's not immediately clear that we're doing that to satisfy the `staticcheck` linter and that it doesn't actually meaningfully change the allocations behavior.

Reply:
Commented at [L317](https://github.com/qingyang-hu/mongo-go-driver/blob/godriver2021/x/mongo/driver/operation.go#L317), where is the direct source of the `*[]byte`